examples/tts/conf/hifigan/hifigan_data.yaml

```diff
@@ -0,0 +1,133 @@
+# This config contains the default values for training a HiFi-GAN model.
+# If you want to train model on other dataset, you can change config values according to your dataset.
+# Most dataset-specific arguments are in the head of the config file, see below.
+
+name: "HifiGan"
+
+defaults:
+  - sample: ???
```

Comment:
(Let's say I'm new user reading this config)

I don't quite get what this `sample` means neither from name nor from context. What should i put here if want to train a model on my dataset?

Reply:
In the defaults lists, `sample: ???` can only mean it is the name of config file in the `sample` directory. It cannot be anything else, like a string value.

Otherwise, I would like to do something like:

```
sample_rate: ???

defaults:
  - sample: sample_${sample_rate}
  - feature: feature_${sample_rate}
  - model/generator: v1_${sample_rate}
```

But my current understanding is that this is impossible to do.

Or is your point more than I should rename the parameter and config directory name from `sample` to `sample_rate`?

I could also add more comments to try to tell a user what some of the other `???` parameters mean, but I doubt anyone will be able to figure out how to use it regardless without seeing a full example/tutorial.

Reply:
I don't think grouping of these parameters is good in the first place.

Sampling rate is a property of dataset we're building a model for. Other two parameters have more to do with available GPU memory than with signals themselves. I think this is why it is hard to come up with a good name for the group - they don't really belong together. I hope you agree with me that `cfg.sample.sample_rate` looks a bit awkward

I'd ask for renaming `train_n_segments` to `train_n_samples` though, if it doesn't harm backward compatibility.

Reply:
I think of having an easy one line way to group all parameters than depend on the sample rate is a good idea. However the pre-existing config hierarchy and vocoder code does not provide an optimal way to do it.

So if we think this looks too awkward, I will require the user to provide all 3 configs separately in the CLI.

Reply:
Changed